flink-table/flink-table-storage/src/main/java/org/apache/flink/table/storage/filestore/lsm/compact/IntervalPartition.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.storage.filestore.lsm.compact;
+
+import org.apache.flink.table.data.binary.BinaryRowData;
+import org.apache.flink.table.runtime.generated.RecordComparator;
+import org.apache.flink.table.storage.filestore.lsm.SortedRun;
+import org.apache.flink.table.storage.filestore.lsm.sst.SstFileMeta;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.PriorityQueue;
+import java.util.stream.Collectors;
+
+/** Algorithm to partition several sst files into the minimum number of {@link SortedRun}s. */
+public class IntervalPartition {
+
+    List<SstFileMeta> files;
+    RecordComparator keyComparator;
+
+    public IntervalPartition(SortedRun sortedRun, RecordComparator keyComparator) {
+        this.files = sortedRun.files();
+        this.keyComparator = keyComparator;
+    }
+
+    public IntervalPartition(List<SortedRun> sortedRuns, RecordComparator keyComparator) {
+        this.files =
+                sortedRuns.stream().flatMap(r -> r.files().stream()).collect(Collectors.toList());
+        // TODO use a merge sort algorithm instead of sorting this directly
+        files.sort(
+                (o1, o2) -> {
+                    int leftResult = keyComparator.compare(o1.minKey(), o2.minKey());
+                    return leftResult == 0
+                            ? keyComparator.compare(o1.maxKey(), o2.maxKey())
+                            : leftResult;
+                });
+        this.keyComparator = keyComparator;
+    }
+
+    public List<List<SortedRun>> partition() {
```

Comment:
What is the output? Can you add document for this?
What is difference between `partition()` and `partition(metas)`?

Reply:
`partition()` is the public interface. It returns several sections (outer list) of sorted runs (inner list). `partition(metas)` is the inner implementation for a specific section. I'll add some documentation.